test/tests/functional/pbs_support_linux_cgroup_phase1_2.py

```diff
@@ -0,0 +1,229 @@
+        test += ['#PBS -l select=vnode=%s+vnode=%s\n' %
+                 (self.hostA, self.hostB)]
+        test += ['%s -j $PBS_JOBID /bin/sleep 30\n' % pbs_attach]
+        test += ['%s %s.pbspro.com %s /bin/sleep 30\n' %
+                 (pbs_tmrsh, self.hostB, pbs_attach)]
+
+        # Submit a job
+        j = Job(TEST_USER)
+        j.create_script(body=test)
+        check_after1 = int(time.time())
+        check_after2 = check_after1 + 2
+        jid = self.server.submit(j)
+        self.server.expect(JOB, {ATTR_state: 'R'}, id=jid)
+
+        # Check mom logs
+        msg_expected = "event is EXECJOB_PROLOGUE"
+        s = self.momA.log_match(msg_expected, starttime=check_after1,
+                                max_attempts=10)
+        self.assertTrue(s)
+        s = self.momB.log_match(msg_expected, starttime=check_after1,
+                                max_attempts=10)
+        self.assertTrue(s)
+
+        msg_expected = "event is EXECJOB_ATTACH"
+        s = self.momA.log_match(msg_expected, starttime=check_after2,
+                                max_attempts=10)
+        self.assertTrue(s)
+        # Allow time for log message to appear before checking on sister mom
+        time.sleep(31)
+        s = self.momB.log_match(msg_expected, starttime=check_after2,
+                                max_attempts=10)
+        self.assertTrue(s)
+
+    def test_execjob_attach_hook_with_accept(self):
+        """
+        Check that the execjob_attach event type with accept
+        of a hook happen in the sister mom and mom superior;
+        execjob_attach hook returns process id, job and vnode list.
+        """
+
+        hook_body = """
+import pbs
+import os
+import sys
+import time
+
+e = pbs.event()
+pbs.logmsg(pbs.LOG_DEBUG,
+           "printing pbs.event() values ---------------------->")
+if e.type == pbs.EXECJOB_ATTACH:
+   pbs.logmsg(pbs.LOG_DEBUG, "Event is: %s" % ("EXECJOB_ATTACH"))
+else:
+   pbs.logmsg(pbs.LOG_DEBUG, "Event is: %s" % ("UNKNOWN"))
+
+pbs.logmsg(pbs.LOG_DEBUG, "Requestor is: %s" % (e.requestor))
+pbs.logmsg(pbs.LOG_DEBUG, "Requestor_host is: %s" % (e.requestor_host))
+
+# Getting/setting vnode_list
+vn = pbs.event().vnode_list
+
+for k in vn.keys():
+   pbs.logmsg(pbs.LOG_DEBUG, "Vnode: [%s]-------------->" % (k))
+
+pbs.logjobmsg(e.job.id, "PID = %d, type=%s" % (e.pid, type(e.pid)))
+
+if e.job.in_ms_mom():
+        pbs.logmsg(pbs.LOG_DEBUG, "job is in_ms_mom")
+else:
+        pbs.logmsg(pbs.LOG_DEBUG, "job is NOT in_ms_mom")
+
+e.accept()
+"""
+
+        a = {'event': 'execjob_attach', 'enabled': 'True'}
+        self.server.create_import_hook("hook1", a, hook_body)
+
+        self.momA = self.moms.values()[0]
+        self.momB = self.moms.values()[1]
+        self.hostA = self.momA.shortname
+        self.hostB = self.momB.shortname
+        if self.momA.is_cpuset_mom():
+            self.hostA = self.hostA + '[0]'
+        if self.momB.is_cpuset_mom():
+            self.hostB = self.hostB + '[0]'
```

Comment:
The filename contains "cgroup" and you are testing whether the cpuset MoM is being used. The cgroup hook should be disabled when the cpuset MoM is running. It doesn't look like you're testing the cgroup hook at all here, so the filename is misleading.  Rather, you are testing hook event presence and order. Perhaps we should consider a different filename.